Source/Modules/lang.cxx

```diff
@@ -629,24 +629,9 @@ int Language::constantDirective(Node *n) {
 
   if (!ImportMode) {
     Swig_require("constantDirective", n, "name", "?value", NIL);
-    String *name = Getattr(n, "name");
-    String *value = Getattr(n, "value");
-    if (!value) {
-      value = Copy(name);
-    } else {
-      /*      if (checkAttribute(n,"type","char")) {
-         value = NewString(value);
-         } else {
-         value = NewStringf("%(escape)s", value);
-         }
-       */
-      Setattr(n, "rawvalue", value);
-      value = NewStringf("%(escape)s", value);
-      if (!Len(value))
-	Append(value, "\\0");
-      /*      Printf(stdout,"'%s' = '%s'\n", name, value); */
-    }
-    Setattr(n, "value", value);
+    if (!Getattr(n, "value")) {
+      Setattr(n, "value", Copy(Getattr(n, "name")));
```

Comment:
I can change that, but it's an existing bug rather than something I've introduced - here's the existing code I replaced with the above:

```
    String *name = Getattr(n, "name");
    String *value = Getattr(n, "value");
    if (!value) {
      value = Copy(name);
    } else {
      /*      if (checkAttribute(n,"type","char")) {
         value = NewString(value);
         } else {
         value = NewStringf("%(escape)s", value);
         }
       */
      Setattr(n, "rawvalue", value);
      value = NewStringf("%(escape)s", value);
      if (!Len(value))
        Append(value, "\\0");
      /*      Printf(stdout,"'%s' = '%s'\n", name, value); */
    }
    Setattr(n, "value", value);
```

When `Getattr(n, "value")` returns `NULL` this boils down to:

```
    String *name = Getattr(n, "name");
    value = Copy(name);
    Setattr(n, "value", value);
```

Reply:
Fixed in 09cecc39e987c1c45bed8ce579488a0dea4d378b (I'll push with a batch of updates)